backend/de.metas.cucumber/src/test/resources/de/metas/cucumber/features/salesOrder.feature

```diff
@@ -30,3 +33,51 @@ Feature: sales order
+      | 40             | 30                 | DE                        | EUR                 | price_list_name_3 | null            | false | false         | 2              | true         |
+    And metasfresh contains M_PriceList_Versions
+      | M_PriceList_Version_ID | M_PriceList_ID | Name                | ValidFrom  |
+      | 30                     | 30             | salesOrder-PLV_2    | 2021-04-01 |
+      | 40                     | 40             | purchaseOrder-PLV_2 | 2021-04-01 |
+    And metasfresh contains M_ProductPrices
+      | M_ProductPrice_ID | M_PriceList_Version_ID | M_Product_ID.Identifier | PriceStd | C_UOM_ID.X12DE355 | C_TaxCategory_ID.InternalName |
+      | 30                | 30                     | p_2                     | 10.0     | PCE               | Normal                        |
+      | 40                | 40                     | p_2                     | 10.0     | PCE               | Normal                        |
+    And metasfresh contains C_BPartners:
+      | Identifier    | Name          | OPT.IsVendor | OPT.IsCustomer | OPT.M_PricingSystem_ID |
+      | endcustomer_2 | Endcustomer_2 | N            | Y              | 30                     |
+      | vendor_1      | vendor_1      | Y            | Y              | 30                     |
+    And metasfresh contains C_BPartner_Products:
+      | C_BPartner_ID.Identifier | M_Product_ID.Identifier |
+      | vendor_1                 | p_2                     |
+    And metasfresh contains C_Orders:
+      | Identifier | IsSOTrx | C_BPartner_ID.Identifier | DateOrdered |
+      | o_2        | true    | endcustomer_2            | 2021-04-17  |
+    And metasfresh contains C_OrderLines:
+      | Identifier | C_Order_ID.Identifier | M_Product_ID.Identifier | QtyEntered |
+      | ol_2       | o_2                   | p_2                     | 10         |
+    And the order identified by o_2 is completed
+    And after not more than 10s, M_ShipmentSchedules are found:
+      | Identifier | C_OrderLine_ID.Identifier | IsToRecompute |
+      | s_ol_2     | ol_2                      | N             |
+    When generate PO from SO is invoked with parameters:
+      | C_BPartner_ID.Identifier | C_Order_ID.Identifier | PurchaseType |
+      | vendor_1                 | o_2                   | Mediated     |
+    Then the order is created:
+      | Link_Order_ID.Identifier | IsSOTrx | DocBaseType | DocSubType |
+      | o_2                      | false   | POO         | MED        |
+    And the mediated purchase order linked to order 'o_2' has lines:
+      | QtyOrdered | LineNetAmt | M_Product_ID.Identifier |
+      | 10         | 100        | p_2                     |
+    And the sales order identified by 'o_2' is closed
+    And the shipment schedule identified by s_ol_2 is processed
```

Comment:
The shipment-schedule's IsProcessed flag is updated asynchronously via `UpdateInvalidShipmentSchedulesWorkpackageProcessor`, so i think this is way too flacky.
Please wait for some time. See `Then after not more than 10s, M_ShipmentSchedules are found:` in line 33 of this file.

Reply:
right, on it